llvm/lib/CAS/HashMappedTrie.cpp

```diff
@@ -467,7 +454,14 @@ void ThreadSafeHashMappedTrieBase::destroyImpl(function_ref<void (void *)> Destr
   // Destroy the subtries using explicit iteration here to avoid recursive
   // destructor calls. Incidentally, this destroys them in reverse order of
   // creation (breadth-first, deepest first).
-  std::unique_ptr<TrieSubtrie> Trie = Impl->Root.Next.take();
-  while (Trie)
-    Trie = Trie->Next.take();
+  TrieSubtrie *Trie = Impl->Root.Next;
+  while (Trie) {
+    TrieSubtrie *Tmp = Trie;
+    Trie = Trie->Next;
+    delete Tmp;
+  }
```

Comment:
The comment above about "recursive destructor calls" seems out of date. There are no RAII destructor calls to avoid anymore.

Also, I think I'd find this easier to read like this:
```
TrieSubtrie *Trie = Impl->Root.Next;
while (Trie) {
  TrieSubtrie *Next = Trie->Next.exchange(nullptr);
  delete Trie;
  Trie = Next;
}
```
This would also allow you to add an assertion to the `TrieSubtrie` destructor that `Next` is nullptr, to confirm it has already been cleaned up.